src/Repository/UserRepository.php

```diff
@@ -115,7 +115,7 @@ public function findLatestContributions(User $user, int $limit = 25) {
 SELECT JSON_AGG(id) AS ids, type FROM (
         SELECT id, timestamp, 'comment'::TEXT AS type FROM comments WHERE user_id = :user_id
```

Comment:
This query needs to be modified so that it doesn't pull comments from mod threads. 

OR

I can change the comment entity to have a modThread value of it's own. 

What do you guys think?

Reply:
DB wise - this looks perfectly good with the mod_thread = false in the where clause.  In fact, that's a better idea if you would basically be duplicating a table so it would have everything else exactly the same, but just a different name.  Use the mod_thread column for this.

My change suggestion is actually that you not call that column "type".  That's a reserved word and can bite you later.  I would switch that to entry_type or something else.

Reply:
Oh!  If the question is "do I join or add a column"?  The answer is almost always "do the join".  This is a relational DB - relations mean something and you should never have the same data in 2 places.  Do you need help writing that query?  Add the join and if it is properly indexed the performance penalty will be minimal.

Also - I don't think your limit will do what you think it will.  I think you are only going to ever have 2 rows there, so that limit of 25 doesn't make any sense (did I read the ORM property correctly there?).

Reply:
The big question here is that this modThread field on the comment entity will probably only get used here, which is for the users page (shows users most recent submissions and comments)

Would adding that field (and it's associated calls) to the comments table be more detrimental in the big picture if it only assists us here?

Reply:
I would love help writing the query! 

This entire query was written by our dear friend, so I'm not sure what the purpose of the limit is here..

The purpose of this PR is for security around mod posts and their comments. If this query needs an overhaul from an efficiency standpoint, let's create a ticket and I'm happy to do any needed code monkey work if you help with the actual query!

Reply:
We have the mod_thread column on the submission table.  And a submission is really a thread/deal.  Then the comments are posted to one submission (it's a child table in my lingo).  We should not add a mod_thread to comment - we should use the one that already exists on the submission table.  So:

```
SELECT c.id, 'comment'::TEXT AS entry_type 
FROM comments c
	INNER JOIN submissions s on c.submission_id = s.id
WHERE c.user_id = :user_id
	and s.mod_thread = false
```

I might have a column name wrong - I'm not at a place where I can actually check the DB right now.

Reply:
It's been updated with that inner join and works like a charm!

I went through and changed all the "type"s everywhere to "entry_type" as well